server/src/test/java/org/apache/druid/segment/indexing/granularity/UniformGranularityTest.java

```diff
@@ -358,6 +359,18 @@ public void testUniformGranularitySpecWithLargeNumberOfIntervalsDoesNotBlowUp()
     Assert.assertEquals(3600 * 24 * 365 * 10 + 3 * 24 * 3600, count);
   }
 
+  @Test(expected = DruidException.class)
+  public void testCheckGranularityValidity()
+  {
+    new UniformGranularitySpec(
+        Granularities.SECOND,
```

Comment:
If there's not a test for it already, should we add the boundary case where segment and query granularities are the same?

Reply:
Added a test where I Assert.fail() in the catch block. Could you please let me know if that works?

Reply:
Thanks for adding it. Regarding the assertion, added a comment here - https://github.com/apache/druid/pull/16324#discussion_r1576913817